AppinfoJsonFile.js

```diff
@@ -0,0 +1,309 @@
+    var fullPath = "";
+
+    var splitLocale = locale.split("-");
+    if (this.baseLocale) {
+        fullPath = "/" + splitLocale[0];
+    } else {
+        for (var i=0; i < splitLocale.length; i++) {
+            fullPath += "/"+ splitLocale[i];
+        }
+    }
+    return rootPath + fullPath;
+};
+
+/**
+ * Localize the text of the current file to the given locale and return
+ * the results.
+ *
+ * @param {TranslationSet} translations the current set of translations
+ * @param {String} locale the locale to translate to
+ * @returns {String} the localized text of this file
+ */
+AppinfoJsonFile.prototype.localizeText = function(translations, locale) {
+    var output = {};
+    for (var property in this.parsedData) {
+        if (this.schema[property]){
+            var text = this.parsedData[property];
+            var key = this.makeKey(this.API.utils.escapeInvalidChars(text));
+            var tester = this.API.newResource({
+                resType: "string",
+                project: this.project.getProjectId(),
+                sourceLocale: this.project.getSourceLocale(),
+                reskey: key,
+                datatype: this.datatype || "javascript"
+            });
+            var hashkey = tester.hashKeyForTranslation(locale);
+            var translated = translations.getClean(hashkey);
+
+            if (translated) {
+                output[property] = translated.target;
+            } else {
+                logger.trace("New string found: " + text);
+                var r = this.API.newResource({
+                    resType: "string",
+                    project: this.project.getProjectId(),
+                    sourceLocale: this.project.getSourceLocale(),
+                    source: this.API.utils.escapeInvalidChars(text),
+                    targetLocale: locale,
+                    target: this.API.utils.escapeInvalidChars(text),
+                    reskey: key,
+                    state: "new",
+                    datatype: this.datatype || "javascript"
+                });
+                this.type.newres.add(r);
+            }
+        }
+    }
+    return output;
+}
+
+/**
+  * Localize the contents of this HTML file and write out the
+  * localized HTML file to a different file path.
+  *
+  * @param {TranslationSet} translations the current set of
+  * translations
+  * @param {Array.<String>} locales array of locales to translate to
+  */
+AppinfoJsonFile.prototype.localize = function(translations, locales) {
+    // don't localize if there is no text
+    for (var i=0; i < locales.length; i++) {
+       if (!this.project.isSourceLocale(locales[i])) {
+            var pathName = this.getLocalizedPath(locales[i]);
+            var translatedOutput = this.localizeText(translations, locales[i]);
+            this.API.utils.makeDirs(pathName);
+            fs.writeFileSync(pathName + "/appinfo.json", JSON.stringify(translatedOutput, true, 4), "utf-8");
```

Comment:
The stringify should be done in the localizeText() call

Reply:
I fixed it